dandi/tests/test_metadata.py

```diff
@@ -513,6 +513,12 @@ def test_species():
     }
 
 
+def test_extract_unknown_species():
+    with pytest.raises(ValueError) as excinfo:
+        extract_species({"species": "mumba-jumba"})
+    assert str(excinfo.value).startswith("Cannot interpret")
```

Comment:
Wouldn't you want this to be:

```suggestion
    assert str(excinfo.value).startswith("Cannot interpret species field: mumba-jumba")
```

Otherwise you're not testing this PR's change

Reply:
this PR change is mostly to do raise expected exception instead of some `UnboundLocalError` or `NameError` but sure -- we can test "deeper".